app/models/pair_history.rb

```diff
@@ -51,6 +51,20 @@ def self.number_of_times_paired_by_name(person_ids)
     end
   end
 
+  def self.pair_history_per_person(team)
```

Comment:
I tried to simplify this a bit. Not sure how much better it is. 

    # example return:
    # {"b"=> {"b"=>0, "a"=>1, "solo"=>0, "out_of_office"=>0},
    #  "a"=> {"b"=>1, "a"=>0, "solo"=>0, "out_of_office"=>0}}
    def self.pair_matrix_per_person(team)
      ids = team.team_members.pluck(:id)
      ungrouped_pairs = number_of_times_paired_by_name(ids)
      team_member_names = team.members_including_statuses.pluck(:name)

      members_hash_with_zero_values = Hash[team_member_names.map {|x| [x, 0] }]
      pairs_by_person = team_member_names.each_with_object({}) do |name, memo|
        memo[name] = members_hash_with_zero_values.dup
      end

      ungrouped_pairs.each_pair do |(name_one, name_two), count|
        pairs_by_person[name_one][name_two] = count
        pairs_by_person[name_two][name_one] = count
      end

      pairs_by_person
    end